server/service/src/stocktake/update.rs

```diff
@@ -589,20 +581,22 @@ fn generate(
             )?
         } else {
             // create new stock line
-            generate_new_stock_line(
-                connection,
-                &store_id,
-                &inventory_addition_id,
-                stocktake_line,
-            )?
+            generate_new_stock_line(&store_id, &inventory_addition_id, &stocktake_line)?
         };
-        stock_lines.push(stock_line);
-        if let Some(shipment_line) = invoice_line {
-            if shipment_line.r#type == InvoiceLineType::StockIn {
-                inventory_addition_lines.push(shipment_line)
-            } else {
-                inventory_reduction_lines.push(shipment_line)
+        match stock_in_out_or_update {
+            Some(StockChange::StockIn(line)) => {
+                inventory_addition_lines.push(line);
+            }
+            Some(StockChange::StockOut(line)) => {
+                inventory_reduction_lines.push(line);
             }
+            Some(StockChange::StockUpdate(stock_line)) => {
+                stock_lines.push(stock_line);
+            }
+            None => {}
```

Comment:
Didn't really want this to be an Option... but needed a way to early exit when it's a new batch, but # packs is 0 (i.e. would be introducing new stock line with 0 quantity). Probably would be better to refactor earlier logic so that's not a possible state by this point, but seems like a separate issue.

Reply:
I actually don't mind what you have here, maybe there is more optimal way to write this, but I think ok for now (a comment could help though, for those in the future that are inclined to try and refactor)